KOGITO-987 - VsCode editor - process id not saved/updated

Josephblt opened the pull request:
https://issues.redhat.com/browse/KOGITO-987

This fix uses the recent added capability of "flushing" added in this JIRA:
[KOGITO-1219](https://issues.redhat.com/browse/KOGITO-1219). The flush method is called each time the document content is requested.
Providing updated content each time.

Josephblt commented:
Jenkins execute full downstream build

Josephblt commented:
Jenkins execute full downstream build

Josephblt commented:
Jenkins please retest this

Josephblt commented:
Jenkins execute full downstream build

Josephblt commented:
Jenkins execute full downstream build

Josephblt commented:
Jenkins execute full downstream build

romartin commented:
Jenkins execute full downstream build

romartin commented on kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-forms/kie-wb-common-stunner-forms-client/src/main/java/org/kie/workbench/common/stunner/forms/client/widgets/FormsFlushManager.java:
```diff
@@ -0,0 +1,45 @@
+
+    protected FormsContainer container;
+
+    public void setCurrentContainer(FormsContainer container) {
```
this method should not be `public`, otherwise doesn't make sense the arguments for the `flush(ClientSession, String)` method, as each session (editor) will have a different associated `FormsContainer` instance, which could not be the one set in  `setCurrentContainer`...

Can you please change it to package protected access (so no access modifier set)? Does it work/make sense for you?

Josephblt commented:
Jenkins execute full downstream build

Josephblt commented:
Jenkins execute full downstream build

romartin commented:
Hey @LuboTerifaj , 

I think you're facing with [KOGITO-1560](https://issues.redhat.com/browse/KOGITO-1560) / [KOGITO-1425](https://issues.redhat.com/browse/KOGITO-1425) and so it seems the PR is not working...

Could you please try to click on the canvas first, before doing anything else, and then feel free to open the properties panel and save whenever you want... it should then work properly.

Anyway I'm testing locally some PRs at same time, including this, and I'll let you konw guys.

Josephblt commented:
Jenkins execute full downstream build

romartin commented:
Hey @LuboTerifaj @Josephblt 

Thanks for the updates! After the new commit from @Josephblt here, I've been testing this, and I can approve it again as it worked for me, although found some issue yet, see [this comment](https://github.com/kiegroup/kie-wb-common/pull/3246#issuecomment-604209768), but we can report it and fix it on top of this. WDYT??

Thanks!

Josephblt commented:
Jenkins execute full downstream build

Josephblt commented:
Jenkins please retest this

LuboTerifaj commented:
Hi @Josephblt, @romartin, 

It is still not possible to save any changes in Properties panel.

If it is blocked by https://github.com/kiegroup/kie-wb-common/pull/3246, please let me know when it is merged and this one is rebased.
In the other case, let me know once the PR is updated with the fix and it is ready to review again.

Thanks!

romartin commented:
ok @LuboTerifaj , lemme ask for urgent review for https://github.com/kiegroup/kie-wb-common/pull/3239

@Josephblt once the PR is merged you'll have to rebase from master and re-fire the builds please.

Thanks!

Josephblt commented:
Jenkins please retest this

Josephblt commented:
Jenkins execute full downstream build

romartin commented:
Jenkins execute full downstream build

romartin commented:
@LuboTerifaj this is rebased from master, so ready for testing. We detected an issue on top of this, although not caused by this, which @Josephblt already reported as well, so if you face it, please notice it's still reported - see [KOGITO-1586](https://issues.redhat.com/browse/KOGITO-1586). Rather than this, it should be everything ok.

Thx!

Josephblt commented:
Jenkins execute full downstream build

LuboTerifaj commented:
Hello @Josephblt , @romartin ,

I tried it again, but I can't see any changes.

It is still not possible to save any changes in properties panel.
I'm sure the process is saved since the changes on the canvas are saved.

romartin commented:
Hey @LuboTerifaj @Josephblt 

I've been testing again and it works for me, see [my comment](https://issues.redhat.com/browse/KOGITO-987?focusedCommentId=14011884&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14011884) about it. @LuboTerifaj are we missing some bug you just found? :)

Thx!

LuboTerifaj commented:
Hey @romartin , 

I've already reported the issue and it can be found here:
https://issues.redhat.com/browse/KOGITO-1621

Thanks!

romartin commented:
please @Josephblt update this PR by rebasing on top of master and let @LuboTerifaj know when it's ready to test again. Thanks!

Josephblt commented:
Jenkins execute full downstream build

Josephblt commented:
Hello @romartin , @LuboTerifaj. Rebased the PR. But from my tests, the latest changes fixed the CTRL+S issue. So, all this does is introduce the "validation" problem. Still, I updated it to save us time, just in case. But I don't think we should merge this. If you guys could please test it and report your findings that would be great. Thanks

romartin commented:
Hey @LuboTerifaj 
ready to test again please! ( Also remember we're still working on [KOGITO-1621](https://issues.redhat.com/browse/KOGITO-1621), which happens on top of this )

LuboTerifaj commented:
Hi @romartin , what do you mean by  "... KOGITO-1586, which happens on top of this"?

romartin commented:
I mean that this (actual) PR is providing the `Ctrl+S` support in VSCode by flushing the forms state, capability that was not being used before (forms recently added flushing capabilities). The downside of this is that now, when saving, all form field validatoins are being fired, and it causes some confusion - so we reported this into [KOGITO-1586](https://issues.redhat.com/browse/KOGITO-1586), and @Josephblt is working on it. But it only happens after tjhe changes from this PR. Does that makes sense? It's just for letting you know, that most probably is you facing also with KOGITO-1586, but it's already reported :)

Thx!

Josephblt commented:
Hello @LuboTerifaj. That actually is related to [KOGITO-1586](https://issues.redhat.com/browse/KOGITO-1586). After the validation fails you have to close and re-open properties tab to be able to edit it again.

LuboTerifaj commented:
Hi @Josephblt ,

I'm not sure what issue you have commented on in my previous [comment](https://github.com/kiegroup/kie-wb-common/pull/3239#pullrequestreview-387355061).

It looks like KOGITO-1586 is related to the first issue, not the second one.

Can you please clarify it?

Thanks!

Josephblt commented:
Hello @LuboTerifaj.

Yeah. Both these problems are related but are not the same. 

KOGITO-1586 is about the fact you can't make changes without reopening the properties tab after any validation fails, not only after the package field one. 

So we should create a new JIRA reporting the fact package field should have some default value,
and fix the general validation issues in KOGITO-1586.

romartin commented:
great, thanks guys, merging!
